src/main/scala/nl.knaw.dans.easy.deposit/StateManager.scala

```diff
@@ -23,51 +23,55 @@ import nl.knaw.dans.easy.deposit.Errors.{ IllegalStateTransitionException, Inval
+case class StateManager(draftDeposit: DepositDir, submitBase: File, easyHome: URL) extends DebugEnhancedLogging {
 
+  private val depositDir = draftDeposit.bagDir.parent
+  private val relativeDraftDir = s"${ draftDeposit.user }/${ draftDeposit.id }"
   private val stateDescriptionKey = "state.description"
   private val stateLabelKey = "state.label"
 
   /** created/destroyed when the state changes, needed to peek into SUBMITTED/UUID/deposit.properties */
   private val bagIdKey = "bag-store.bag-id"
 
   /** Available as side effect for properties not related to fetching or updating the state of the deposit. */
+  @throws[ConfigurationException]
   val draftProps = new PropertiesConfiguration(
     (depositDir / "deposit.properties").toJava
   )
-  private lazy val submittedProps = new PropertiesConfiguration(
-    (submitBase / getProp(bagIdKey, draftProps) / "deposit.properties").toJava
-  )
+  private lazy val submittedProps = getProp(bagIdKey, draftProps)
+    .map(submittedId =>
+      new PropertiesConfiguration((submitBase / submittedId.toString / "deposit.properties").toJava)
+    ).getOrElse(new PropertiesConfiguration)
 
   /** @return the state-label/description from drafts/USER/UUID/deposit.properties
    *          unless more recent values might be available in SUBMITTED/UUID/deposit.properties
    */
-  def getStateInfo: Try[StateInfo] = Try {
-    getStateInDraftDeposit match {
-      case stateInDraftDeposit @ (State.submitted | State.inProgress) =>
-        val stateInSubmittedDeposit: StateInfo = Try(getProp(stateLabelKey, submittedProps)) match {
-          case Success("SUBMITTED") => StateInfo(State.submitted, getStateDescription(draftProps))
-          case Success("REJECTED") => StateInfo(State.rejected, getStateDescription(submittedProps))
-          case Success("FAILED") => StateInfo(State.inProgress, s"The deposit is in progress.")
-          case Success("IN_REVIEW") => StateInfo(State.inProgress, s"""The deposit is available at <a href="$landingPage" target="_blank">$landingPage</a>""")
-          case Success("FEDORA_ARCHIVED") |
-               Success("ARCHIVED") => StateInfo(State.archived, s"""The dataset is published at <a href="$landingPage" target="_blank">$landingPage</a>""")
-          case Success(str: String) =>
-            logger.error(InvalidPropertyException(stateLabelKey, str, submittedProps).getMessage)
-            StateInfo(State.inProgress, s"The deposit is in progress.")
-          case Failure(e) =>
-            logger.error(e.getMessage, e)
-            StateInfo(stateInDraftDeposit, getStateDescription(draftProps))
+  def getStateInfo: Try[StateInfo] = getStateInDraftDeposit.map {
+    case draftState @ (State.draft | State.rejected | State.archived) =>
+      StateInfo(draftState, getStateDescription(draftProps))
+    case stateInDraftDeposit @ (State.submitted | State.inProgress) =>
+      getProp(stateLabelKey, submittedProps) match {
```

Comment:
maybe outside the scope of this issue, but a `.map` with pattern match would be less repetitious that this match with `case Success(...)`

Reply:
Also a match on enum values would maybe be more desirable than matching on strings?

Reply:
Ideally the enum should somehow be shared with ingest-flow. I doubt we have a good mechanism for that purpose.

Reply:
Not sure if it's a good idea to have a shared enum (a.k.a. a library) for this. Hopefully in the near future we will move this to `easy-deposit-properties`, whose GraphQL interface **guarantees** a specific set of values.